### PR TITLE
refactor(coq): refactor coqdep parsing

### PR DESCRIPTION
**Pull request by Alizter**
This is some preliminary refactoring while I'm working on vos support.

### CODE REVIEW

File: src/dune_rules/coq_rules.ml [hunk 1]
```diff
@@ -370,27 +370,31 @@ module Context = struct
     { t with boot_type }
 end
 
-let parse_coqdep ~dir ~(boot_type : Bootstrap.t) ~coq_module
-    (lines : string list) =
+let parse_coqdep ~dir ~(boot_type : Bootstrap.t) ~coq_module contents =
```
Comment (ejgallego) [inline src/dune_rules/coq_rules.ml:373]: `lines` is a better name than `contents`, moreover when this is called from `lines_of`

File: src/dune_rules/coq_rules.ml [hunk 2]
```diff
@@ -370,27 +370,31 @@ module Context = struct
...
+          , String.extract_blank_separated_words deps ))
   in
-  match String.lsplit2 line ~on:':' with
-  | None -> invalid "split"
+  let ext = ".vo" in
+  match
+    List.find_map deps ~f:(fun (targets, deps) ->
+        List.find_map targets ~f:(fun t ->
+            if Filename.check_suffix t ext then Some (t, deps) else None))
+  with
```
Comment (ejgallego) [inline src/dune_rules/coq_rules.ml:396]: Building the (inefficient) map just to deconstruct it is wasted work IMO, moreover this code is less resilient to changes in coqdep.

Comment (ejgallego) [reply src/dune_rules/coq_rules.ml:396]: Coqdep parsing has two modes:
- single file shot: we just need to read the first line and the first target, the rest are useless; we output a list of files
- multi-file shot: we first need to filter relevant lines, then build a (true) map with the module deps

**Comment by Alizter** [comment]
Abandoning this since it is actually not needed. We can infer vos deps ourselves without having to reread coqdep.

**Pull request closed without merging**